--- lirc-0.9.0/Makefile.in	2011-03-25 23:28:54.000000000 +0100
+++ lirc-0.9.0_new/Makefile.in	2014-01-30 23:38:07.415594059 +0100
@@ -252,7 +252,7 @@
 x_progs = @x_progs@
 AUTOMAKE_OPTIONS = 1.5 check-news dist-bzip2
 EXTRA_DIST = setup.sh setup-driver.sh setup.data data2setup.sh ANNOUNCE TODO
-SUBDIRS = drivers daemons tools doc
+SUBDIRS = daemons tools doc
 DISTCHECK_FLAGS = --with-driver=all --enable-sandboxed
 DISTCHECK_CONFIGURE_FLAGS = ${DISTCHECK_FLAGS}
 DISTCHECK_LIGHT_FLAGS = --with-driver=none --enable-sandboxed
--- lirc-0.9.0/Makefile.am	2011-03-25 23:28:18.000000000 +0100
+++ lirc-0.9.0_new/Makefile.am	2014-01-30 23:37:52.965587075 +0100
@@ -9,7 +9,7 @@
 EXTRA_DIST =  setup.sh setup-driver.sh setup.data data2setup.sh ANNOUNCE TODO
 
 ## daemons/ and tools/ should appear before doc/ for help2man to work.
-SUBDIRS = drivers daemons tools doc
+SUBDIRS = daemons tools doc
 
 ## From automake documentation:
 ## Note that EXTRA_DIST can only handle files in the current
